ProcessQueue/__init__.py

```diff
@@ -266,6 +266,302 @@ def process_failure(api_url, scan_report_id, headers):
         headers=headers
     )
 
+def paginate_chars(entries_to_post,other):
+    """
+    This expects a list of dicts, and returns a list of lists of dicts, 
+    where the maximum length of each list of dicts, under JSONification, 
+    is less than max_chars
+    """
+    max_chars = 2000 - len(other)
+    
+    paginated_entries_to_post = []
+    this_page = []
+    for entry in entries_to_post:
+        # If the current page won't be overfull, add the entry to the current page
+        if len(json.dumps(this_page)) + len(
+                json.dumps(entry)) < max_chars:
+            this_page.append(entry)
+        else:
+            # Otherwise, this page should be added to the list of pages.
+            paginated_entries_to_post.append(this_page)
+            # Now add the entry that would have over-filled the page.
+            this_page = [entry]
+
+    # After all entries are added, check for a half-filled page, and if present add it to the list of pages
+    if this_page:
+        paginated_entries_to_post.append(this_page)
+    return paginated_entries_to_post
+
+def flatten(arr):
+    """
+    This expects a list of lists and returns a flattened list
+    """
+    newArr = [item for sublist in arr for item in sublist]
+    return newArr
+
+def get_existing_field_concepts(new_field_ids,content_type,api_url,headers):
```

Comment:
Sorry to flip-flop about what this should be called. Please change `new_field_ids` to `new_fields_map`, and then in the below, `names_string` -> `new_fields_names_string`, and remove the intermediate variable `names_list`. Where it gets reused on line 375, just use `new_fields_map.keys()` again. I think that is then clearer to follow.

Reply:
Also, please rename this function to `reuse_existing_field_concepts`, and similarly for the values function.

Reply:
done

Reply:
Feel free to hit the "Resolve conversation" button rather than commenting when each is completed.